app/views/solutions/edit.html.haml

```diff
@@ -0,0 +1,8 @@
+.md-container
+  Really sexy editor for #{@track.title}: #{@exercise.title}
+
+  = form_with url: api_solution_path(@solution.uuid), method: :patch do
+    = hidden_field_tag :auth_token, current_user.auth_tokens.first.token
+    = hidden_field_tag :major, false
+    = text_area_tag "files[foobar.rb]", "class Code\nend", style: 'border:2px solid black;padding:20px;'
+    = button_tag "Submit"
```

Comment:
@kntsoriano This is where the editor component should be rendered from, and this form submits something, so you can replicate in JS.